src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -664,7 +666,7 @@ public static boolean isLivingOnLadder(@Nonnull IBlockState state, @Nonnull Worl
                     {
                         BlockPos tmp = new BlockPos(x2, y2, z2);
                         state = world.getBlockState(tmp);
-                        if (state.getBlock().isLadder(state, world, tmp, entity))
+                        if (state.getBlock().isLadder(state, world, tmp, entity, side))
```

Comment:
Ok, so here are you only calling the new implementation - you either have to make your new implementation call through to the old method, or use both in order to check